### ISSUE TITLE
Simplify configuration and improve "getting started" documentation

### ISSUE THREAD

**Issue by Place1** [Place1/wg-access-server#43]
A big goal of mine is to keep wg-access-server as simple to get started with as possible. I wrote this software in the first place because I always struggled to get OpenVPN and other solutions up and running.

Today wg-access-server is configured with a config file primarily and then some flags/envvars. I'd like to remove the need for a config file entirely and improve the documentation to focus on running the server with environment variables as config.

The 2 primary supported ways to run this software are standalone docker and kubernetes via helm; both of which make configuration via environment variables simple.

I'm planning to follow the general approach that grafana takes: https://grafana.com/docs/grafana/latest/installation/configuration/

In this issue i want to reach out to anyone using wg-access-server for their thoughts on config + documentation.

cc @halkeye @KimchaC

**Comment by Place1** [comment]
I'm particularly interested in making it easy to configure auth backends using envvars only!

**Comment by halkeye** [comment]
I'll try and take a look this weekend, got a new job last week and not used to the schedule

**Comment by Fastidious** [comment]
I am wanting to run, and use this, but I am not sure how to add the things I want. I would like to:

- Set `cidr`.
- Set the following `rules`: 
    - `allowVPNLAN: true`
    - `allowServerLAN: true`
    - `allowInternet: true`
- Save configurations, and user data on a permanent volume.

How to accomplish that on a docker command? Simply tagging along here, because I believe it relates, but if needed be I will open a separate issue.

**Comment by halkeye** [comment]
The grafana approach for config isn't that bad if you know what your doing, but knowing the config file format is kinda necessary to know which envs variable names to use. It does make managing arrays hard, but all key value and simple string/ints are pretty easy.

Key thing to remember.
1) All env variables are strings, you would have to map them
2) multi word keys become complicated. do you seperate each level with __? or use __ to mean literal _? MAIN__ALLOW_VPN_LAN?

Personally i think if you have sane defaults, and only need to override as necessary, yaml is fine. I like ENV for a few variables, but if i have 10+ I don't want to use env. Then again I use k8s with its stateless configmap so it matters less to me.

**Comment by Place1** [comment]
@halkeye yeah, you make a good point and i agree.

perhaps we should do our best with _sane defaults_ and support a single "CONFIG" environment variable that can be a path to a config file or the inline yaml content itself.

example:

```bash
docker run \
  # ... \
  -v $(pwd):/config \
  -e CONFIG=/config/config.yaml \
  place1/wg-access-server
```

```bash
CONFIG="
  rules:
    allowVPNLAN: false
  wireguard:
    privateKey: "abc123"
"

docker run \
  # ... \
  -e "CONFIG=$CONFIG" \
  place1/wg-access-server
```

**Comment by halkeye** [comment]
Why the same config var do 2 different things? Why not CONFIG_FILE and CONFIG
and I wonder what the max length of env variable is

**Comment by Place1** [comment]
Yeah we can make them different variables if you like.

Env vars can be pretty big so it should be fine.

**Comment by gmag11** [comment]
I've written a post showing how to install wg access server. It's in Spanish, but using a translator you can get something readable. You may get whatever you find useful there. https://blog.gmprojects.pro/posts/intalar-wireguard-con-docker-y-wg-access-server/

**Comment by sentriz** [comment]
there is also https://github.com/kelseyhightower/envconfig

which lets you do stuff like nested stucts like `Wireguard.InterfaceName` or `Wireguard.ExternalHost` (which i think would translate to `WIREGUARD_INTERFACE_NAME` and `WIREGUARD_EXTERNAL_HOST`)
and it can do slices say for `Wireguard.Rules.AllowedNetworks`

**Comment by ThellraAK** [comment]
I said it in another bug report, so I'm sorry for being redundant, it should be more clear in the quick start stuff that if you don't supply a private key, clients can't be persistent through reboot, I restart a lot poking around, so I was able to figure out that was what was going on, but if someone were to install it, use it for a few months or something, it wouldn't be super apparent that is what was happening. 

a way for it to generate it's own private key, and have it persist somewhere outside the config would be really handy.

**Comment by Place1** [comment]
Today i've published v0.3.0 and it includes configuration changes to cover the ideas from this issue.

Config is overall more consistent now. See the changelog for more details.

Thanks everyone for helping me here :D
